src/main/groovy/com/bmuschko/gradle/docker/DockerConventionApplicationPlugin.groovy

```diff
@@ -190,7 +190,11 @@ abstract class DockerConventionApplicationPlugin<EXT extends DockerConventionApp
         })
     }
 
-    private String getApplicationMainClassName(Project project) {
+    private String getApplicationMainClassName(Project project, EXT extension) {
+        if (extension.mainClassName.isPresent()) {
+            return extension.mainClassName.get()
```

Comment:
Should we check to see if the class is actually present or just wait until things fall over at a later time?

Reply:
I think that would only work if the main class can be found in the same Gradle project. If the main lives in a different Gradle project e.g. as a project dependency then finding the main class would fail (see https://github.com/bmuschko/gradle-docker-plugin/issues/893).